### ISSUE TITLE
Microsoft Word, Shift + F3 and F8, NVDA Doesn't Read Selected Text

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#3293]
**Reported by jhomme on 2013-06-20 12:57**
In Microsoft Word, F3 cycles through changing the case of text. NVDA is not speaking the text when the case changes. Steps to reproduce:
1. Create a line of text with several words.
2. Put the cursor on any of the words.
3. Press F3.
4. Word selects the text and changes the case. It would be good for NVDA to read the text, and possibly spell it so that the user can determine how the text case has changed.
5. Select multiple words or the whole line of text.
6. Once again, NVDA does not speak the selected text.
7. I believe that Word also handles larger chunks of text this way, but usually, I use this feature to change the case of single words or lines.
8. This same kind of thing happens when you use F8 to select larger and larger bits of text.

**Comment by leonardder** [comment]
I can reproduce this in Word 2016, at least when I use shift+f3 as in the issue title.

I wonder whether it is possible to track down this command, similar to changing bold with ctrl+b, italic with ctrl+i, etc.

CC @MichaelDCurran, @FeerRenrut

**Comment by ehollig** [comment]
I am not sure if this is what the original poster is talking about, but I can also confirm this in Word 2016.
Shift+F3 = Change the case of letters
F8 = Extend a selection
Shift+F8 =Reduce the size of a selection
https://support.office.com/en-us/article/Keyboard-shortcuts-for-Microsoft-Word-on-Windows-95ef89dd-7142-4b50-afb2-f762f663ceb2

**Comment by Qchristensen** [comment]
This is still reproduceable with:

NVDA 2024.4.1 
Windows 11 (64-bit) Version: 23H2, Build: 22631.4460 
Office 365 (64-bit) Version: 16.0.18227.20046 

Note that a workaround for toggle case is to perform the actions (select text then shift+f3) then press Numpad 5 (NVDA+control+dot) twice quickly to spell the current word) which will also give you the capitalisation.

**Comment by Qchristensen** [comment]
Also worth noting that testing with Narrator (on the same setup) - it re-reads the selected text each time I press shift+f3, but with nothing to indicate what the capitalisation is.  I'm not sure if that's just I haven't set something correctly in Narrator, but I've sent Microsoft a query on this as well.

**Comment by Qchristensen** [comment]
Sorry one more update - it does not matter how NVDA's advanced setting "Use UI Automation to access Microsoft Word document controls" is set.

**Comment by comanna** [comment]
Hi, Quentin,

 

Following your last message, I Tried out Narrator. I set Narrator to report capitalization with Narrator + 4 on the main keyboard, and selecting “say caps”. When I pressed shift + F3, Narrator spoke the character under the cursor and put the word “cap” before the letter if it was capitalised.

 

Hat was not much in itself, but, when I arrowed up and down the text, Narrator spoke the full line and said “All caps” when all letters were capitalised  and only “Cap” before the sentence case. There was no addition when the sentence was all lower case.

 

All the best,

 

Cearbhall

 

m +353 (0)833323487 e: ***@***.***

**Comment by Adriani90** [comment]
The Pr referenced is a partial fix.

**Comment by CyrilleB79** [comment]
@Adriani90 could you be more specific?

This issue was not well written, asking two different thing (`shift+f3` and `f8`). I guess you are referring to `f8`? (but better explicit than implicit)

IMO, rather than reopening this issue, writing a brand new issue with what remains to do would be much better. Could you do this?

**Comment by CyrilleB79** [comment]
@Adriani90, this issue covers various issues.

For clarity, I have opened #17450 to track and clarify what remains to do from this ticket. I'll close this one.

Let me know if I have missed something.
